### PR TITLE
Fix the misleading log messages

### PR DESCRIPTION
**Pull request by derekbit**
[Longhorn 4980](https://github.com/longhorn/longhorn/issues/4980)

Signed-off-by: Derek Su <derek.su@suse.com>

### CODE REVIEW

File: pkg/replica/replica.go [hunk 1]
```diff
@@ -757,12 +758,19 @@ func (r *Replica) rmDisk(name string) error {
 		return nil
 	}
 
+	logrus.Infof("Removing disk %v", name)
+
 	lastErr := os.Remove(r.diskPath(name))
+	if lastErr != nil {
+		logrus.WithError(lastErr).Errorf("Failed to remove dis file %v", name)
```
Comment (innobead) [inline pkg/replica/replica.go:765]: `Failed to remove disk file `

File: pkg/replica/replica.go [hunk 2]
```diff
@@ -757,12 +758,19 @@ func (r *Replica) rmDisk(name string) error {
...
+	}
 	if err := os.Remove(r.diskPath(name + diskutil.DiskMetadataSuffix)); err != nil {
 		lastErr = err
+		logrus.WithError(lastErr).Errorf("Failed to remove disk metadata file %v", name+diskutil.DiskMetadataSuffix)
```
Comment (innobead) [inline pkg/replica/replica.go:769]: Can we do this way?

```go
    diskMetaPath := r.diskPath(name + diskutil.DiskMetadataSuffix)
	if err := os.Remove(diskMetaPath); err != nil {
		lastErr = err
		logrus.WithError(lastErr).Errorf("Failed to remove disk metadata file %v", diskMetaPath)
	}
	
	diskChecksumPath = r.diskPath(name + diskutil.DiskChecksumSuffix)
	if err := os.RemoveAll(diskChecksumPath); err != nil {
		lastErr = err
		logrus.WithError(lastErr).Errorf("Failed to remove disk checksum file %v", diskChecksumPath)
	}
```

**Review by innobead — CHANGES_REQUESTED** [review]
In general, LGTM. Some comments needed to resolve.

File: pkg/replica/replica.go (hunk 2 above)
Comment (derekbit) [reply pkg/replica/replica.go]: Updated. Thank you.

**Review by innobead — APPROVED** [review]

**Pull request merged**